packages/mui-material/src/FormControlLabel/FormControlLabel.test.js

```diff
@@ -18,17 +18,11 @@ describe('<FormControlLabel />', () => {
     render,
     muiName: 'MuiFormControlLabel',
     testVariantProps: { disabled: true },
-    testLegacyComponentsProp: true,
```

Comment:
Is it necessary to remove this? If not, I would remove it when we remove `componentsProps`

Reply:
got it makes sense, restored in this commit -> https://github.com/mui/material-ui/pull/41767/commits/acb25986e979dcf48c9957d62314a57e32e4faf0

Reply:
I was thinking we keep

```
testLegacyComponentsProp: true,
```

Or does that fail?

Reply:
Tests are failing on keeping ot to false.

Reply:
Ok, I see what happened. The `testLegacyComponentsProp` only applies to the slots-related tests. Previously, it was `true`, but the slots-related tests were skipped, so `testLegacyComponentsProp: true` had no effect. As we stop skipping the slots tests, this flag has an effect, and because the component only has `componentsProps` and no `components`, the test fails.

Let's remove `testLegacyComponents: true` as you did initially. It should've never been added and was probably copy-pasted. Sorry for the confusion; my bad.

Reply:
no problem :)
removed `testLegacyComponents`